### PR TITLE
 Updated CodeEvalResult to use time and memory limit exceeded flags.

### PR DESCRIPTION
**Pull request by eyurko**

### CODE REVIEW

File: client/question/domain/CodeEvalResultObjectFactory.js [hunk 1]
```diff
@@ -143,8 +159,7 @@ tie.factory('CodeEvalResultObjectFactory', [
      * @returns {boolean}
      */
     CodeEvalResult.prototype.hasTimeLimitError = function() {
-      var errorString = this.getErrorString();
-      return (errorString && errorString.startsWith('TimeLimitError'));
+      return this.getErrorString() && this._timeLimitExceeded;
```
Comment (seanlip) [inline client/question/domain/CodeEvalResultObjectFactory.js]: Do you actually need the this.getErrorString() bit?

File: client/question/domain/CodeEvalResultObjectFactory.js [hunk 2]
```diff
@@ -154,10 +169,7 @@ tie.factory('CodeEvalResultObjectFactory', [
      * @returns {boolean}
      */
     CodeEvalResult.prototype.hasRecursionLimitError = function() {
-      var errorString = this.getErrorString();
-      return errorString && (
-        errorString.startsWith('ExternalError: RangeError') ||
-        errorString.includes('maximum recursion depth exceeded'));
+      return this.getErrorString() && this._memoryLimitExceeded;
```
Comment (seanlip) [inline client/question/domain/CodeEvalResultObjectFactory.js]: Ditto.

Also maybe rename to hasMemoryLimitError (this is a bit different from recursion limit, which is a subcase). Possibly we should change the error message shown to the user too.

File: client/question/services/code_evaluators/PythonCodeRunnerService.js [hunk 3]
```diff
@@ -127,8 +127,14 @@ tie.factory('PythonCodeRunnerService', [
```
Comment (seanlip) [inline client/question/services/code_evaluators/PythonCodeRunnerService.js]: Don't you need to add a couple more args in line 120 above?

Please double-check calls to CodeEvalResultObjectFactory.create and make sure you've updated all of them.

File: client/question/domain/CodeEvalResultObjectFactory.js (hunk 2 above)
Comment (eyurko) [reply client/question/domain/CodeEvalResultObjectFactory.js]: I didn't update the error message -- what do you have in mind?

"Your code used more memory than we allow for this exercise; please edit your code to be more space-efficient."?

File: client/question/services/code_evaluators/PythonCodeRunnerService.js (hunk 3 above)
Comment (eyurko) [reply client/question/services/code_evaluators/PythonCodeRunnerService.js]: whoops yeah sorry, forgot that Javascript is chill with me just ignoring arguments. Should be fixed, now.

File: client/question/domain/CodeEvalResultObjectFactory.js (hunk 1 above)
Comment (eyurko) [reply client/question/domain/CodeEvalResultObjectFactory.js]: Nope!

File: client/question/domain/CodeEvalResultObjectFactory.js (hunk 2 above)
Comment (seanlip) [reply client/question/domain/CodeEvalResultObjectFactory.js]: Yes, something like that seems reasonable to me. (Though I might replace the semicolon with a period.)

Thanks!

**Pull request merged**